components/script/dom/create.rs

```diff
@@ -76,21 +76,44 @@ use dom::htmltrackelement::HTMLTrackElement;
 use dom::htmlulistelement::HTMLUListElement;
 use dom::htmlunknownelement::HTMLUnknownElement;
 use dom::htmlvideoelement::HTMLVideoElement;
+use dom::svgsvgelement::SVGSVGElement;
 use string_cache::{Atom, QualName};
+use util::prefs::PREFS;
 
-pub fn create_element(name: QualName,
-                      prefix: Option<Atom>,
-                      document: &Document,
-                      creator: ElementCreator)
+fn create_svg_element(name: QualName,
+                      prefix: Option<DOMString>,
+                      document: &Document)
                       -> Root<Element> {
-    // FIXME(ajeffrey): Convert directly from Atom to DOMString.
+    assert!(name.ns == ns!(svg));
 
-    let prefix = prefix.map(|p| DOMString::from(&*p));
+    macro_rules! make(
+        ($ctor:ident) => ({
+            let obj = $ctor::new(name.local, prefix, document);
+            Root::upcast(obj)
+        });
+        ($ctor:ident, $($arg:expr),+) => ({
+            let obj = $ctor::new(name.local, prefix, document, $($arg),+);
+            Root::upcast(obj)
+        })
+    );
```

Comment:
Actually it's not so obvious. If I move macro out of scope I'll lose name.local, prefix, document local vars and should make them arguments. That makes macro mostly useless.